src/sentry/tasks/unmerge.py

```diff
@@ -470,15 +468,34 @@ def update_tag_value_counts(id_list):
         )
 
 
+def lock_hashes(project_id, source_id, fingerprints):
+    with transaction.atomic():
+        eligible_hashes = list(
```

Comment:
This needs to accurately return the fingerprints that were actually locked and you can't reliably do that without a `RETURNING` clause or a higher isolation level without using `SELECT OR UPDATE`.